google/cloud/bigquery/standard_sql.py

```diff
@@ -0,0 +1,207 @@
+# Copyright 2021 Google LLC
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     https://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from dataclasses import dataclass
+from typing import Any, Dict, Iterable, Optional
+
+from google.cloud.bigquery.enums import StandardSqlTypeNames
+
+
+@dataclass
```

Comment:
After seeing how these classes are used elsewhere, I think the StandardSql classes are worth going with our `_properties` pattern.

There's a particular feature that can support, which is that folks can modify the `_properties` dictionary directly when they are testing out a feature in private preview before we add support for it in the client libraries.

Reply:
Even if not the purest solution, _practicality beats purity_ - if users put `_properties` to good use (even at their own risk), then let's preserve that. I'll refactor the standard SQL classes implementation.